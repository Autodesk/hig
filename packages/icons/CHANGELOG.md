# [@hig/icons-v3.4.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.4.0...@hig/icons@3.4.1) (2021-06-26)


### Bug Fixes

* do not spread unknown props to svg ([ee223d1](https://github.com/Autodesk/hig/commit/ee223d1))

# [@hig/icons-v3.4.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.3.0...@hig/icons@3.4.0) (2021-02-22)


### Features

* add stylesheet prop ([decd68a](https://github.com/Autodesk/hig/commit/decd68a))

# [@hig/icons-v3.3.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.2.1...@hig/icons@3.3.0) (2020-05-15)


### Features

* add save-as icons ([197294b](https://github.com/Autodesk/hig/commit/197294b))

# [@hig/icons-v3.2.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.2.0...@hig/icons@3.2.1) (2020-05-12)


### Bug Fixes

* stop using deprecated theme-data roles ([59e91ae](https://github.com/Autodesk/hig/commit/59e91ae))

# [@hig/icons-v3.2.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.1.0...@hig/icons@3.2.0) (2020-05-01)


### Bug Fixes

* remove title element ([8a13814](https://github.com/Autodesk/hig/commit/8a13814))


### Features

* add new icons ([ca0efcd](https://github.com/Autodesk/hig/commit/ca0efcd))

# [@hig/icons-v3.1.0](https://github.com/Autodesk/hig/compare/@hig/icons@3.0.1...@hig/icons@3.1.0) (2019-07-24)


### Features

* distribute SVG source files with [@hig](https://github.com/hig)/icons ([18ac0ed](https://github.com/Autodesk/hig/commit/18ac0ed))

# [@hig/icons-v3.0.1](https://github.com/Autodesk/hig/compare/@hig/icons@3.0.0...@hig/icons@3.0.1) (2019-07-16)


### Bug Fixes

* remove ExternalLinkUI icon ([06fbbde](https://github.com/Autodesk/hig/commit/06fbbde))

# [@hig/icons-v3.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.2...@hig/icons@3.0.0) (2019-07-09)


### Features

* updating icons ([67a908a](https://github.com/Autodesk/hig/commit/67a908a))


### BREAKING CHANGES

* Updated styling of all 16 and 24px icons to better align with existing command icons

* Added add-page
* Added external-link
* Added file-audio
* Added import
* Added inbox
* Added mute
* Added not-editable
* Added select-all
* Added select-none
* Added sort
* Added unlink
* Added unpin
* Added user-inactive
* Renamed building-ops to building
* Renamed cost-control to bill-stack
* Renamed export to export-move
* Renamed insight to graph
* Renamed issue to alert
* Renamed layout to survey-equipment
* Renamed user-incactive to user-inactive
* Removed assets
* Removed caret - can be found in UI Controls
* Removed check-disabled - can be found in UI Controls
* Removed check-white
* Removed checkbox-checked - can be found in UI Controls
* Removed checkbox-indeterminate - can be found in UI Controls
* Removed checkmark-blue-dark
* Removed checkmark-indeterminate - can be found in UI Controls
* Removed checkmark-s - can be found in UI Controls
* Removed clear-small
* Removed close - can be found in UI Controls
* Removed close-hamburger - can be found in UI Controls
* Removed close-notification - can be found in UI Controls
* Removed close-xs - can be found in UI Controls
* Removed dropdown-caret-down - can be found in UI Controls
* Removed dropdown-caret-left - can be found in UI Controls
* Removed dropdown-caret-right - can be found in UI Controls
* Removed dropdown-caret-up - can be found in UI Controls
* Removed hamburger - can be found in UI Controls
* Removed hamburger-close - can be found in UI Controls
* Removed hamburger-default - can be found in UI Controls
* Removed model-coordination
* Removed rfi
* Removed social-blog-2
* Removed x-close-gray

# [@hig/icons-v2.1.2](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.1...@hig/icons@2.1.2) (2019-06-25)


### Bug Fixes

* don't show tooltip on hover ([e50e34f](https://github.com/Autodesk/hig/commit/e50e34f))

# [@hig/icons-v2.1.1](https://github.com/Autodesk/hig/compare/@hig/icons@2.1.0...@hig/icons@2.1.1) (2019-03-13)


### Bug Fixes

* theme-context and theme-data as peer dependencies ([34cd334](https://github.com/Autodesk/hig/commit/34cd334))

# [@hig/icons-v2.1.0](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.2...@hig/icons@2.1.0) (2019-03-08)


### Features

* add ui control icons ([5465341](https://github.com/Autodesk/hig/commit/5465341))

# [@hig/icons-v2.0.2](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.1...@hig/icons@2.0.2) (2019-02-08)


### Bug Fixes

* bump up theme-context dependency ([b068dab](https://github.com/Autodesk/hig/commit/b068dab))

# [@hig/icons-v2.0.1](https://github.com/Autodesk/hig/compare/@hig/icons@2.0.0...@hig/icons@2.0.1) (2019-01-23)


### Bug Fixes

* **package:** update [@hig](https://github.com/hig)/theme-context to version 2.0.0 ([7975948](https://github.com/Autodesk/hig/commit/7975948))

# [@hig/icons-v2.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@1.0.0...@hig/icons@2.0.0) (2019-01-17)


### chore

* **icons:** release icons 2.0.0 ([b61b2cb](https://github.com/Autodesk/hig/commit/b61b2cb))


### Features

* **icons:** make icons themable ([56abadd](https://github.com/Autodesk/hig/commit/56abadd))
* support color attribute on avatar ([c7121d1](https://github.com/Autodesk/hig/commit/c7121d1))


### BREAKING CHANGES

* **icons:** update package version to release icons 2.0.0

# [@hig/icons-v1.0.0](https://github.com/Autodesk/hig/compare/@hig/icons@0.2.1...@hig/icons@1.0.0) (2018-12-04)


### BREAKING

* **icon/icon-button:** Removes [@hig](https://github.com/hig)/icon and refactor [@hig](https://github.com/hig)/icon-button to use [@hig](https://github.com/hig)/icons ([49d78f4](https://github.com/Autodesk/hig/commit/49d78f4))


### Bug Fixes

* **storybook:** Add back Icon stories ([f38f2d4](https://github.com/Autodesk/hig/commit/f38f2d4))


### BREAKING CHANGES

* **icon/icon-button:** Removes @hig/icon
* **icon/icon-button:** Use icon prop instead of name or svg in @hig/icon-button to render Icon/svg
* **icon/icon-button:** No longer export `names` from @hig/icon-button

Usage:
```
import { Assets24 } from "@hig/icons"

function MyComponent() {
  return (
    <IconButton
      icon={<Assets24 />}
      onClick={() => console.log('Clicked it.')}
    />
  );
}
```

# [@hig/icons-v0.2.1](https://github.com/Autodesk/hig/compare/@hig/icons@0.2.0...@hig/icons@0.2.1) (2018-10-08)


### Bug Fixes

* replace stroke icons with fill ([5e06c82](https://github.com/Autodesk/hig/commit/5e06c82))

# [@hig/icons-v0.2.0](https://github.com/Autodesk/hig/compare/@hig/icons@0.1.0...@hig/icons@0.2.0) (2018-08-07)


### Features

* update icon set ([80bf8c6](https://github.com/Autodesk/hig/commit/80bf8c6))
