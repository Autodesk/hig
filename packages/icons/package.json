{
  "name": "@hig/icons",
  "version": "0.1.0",
  "description": "HIG icon assets",
  "author": "Autodesk Inc.",
  "license": "Apache-2.0",
  "homepage": "https://hig.autodesk.com",
  "repository": {
    "type": "git",
    "url": "https://github.com/Autodesk/hig.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "main": "build/index.js",
  "module": "build/index.es.js",
  "files": [
    "build/*"
  ],
  "devDependencies": {
    "@hig/babel-preset": "^0.1.0",
    "@hig/eslint-config": "^0.1.0",
    "@hig/jest-preset": "^0.1.0",
    "@hig/scripts": "^0.1.2",
    "@hig/semantic-release-config": "^0.1.0",
    "@hig/styles": "^0.1.1",
    "svgo": "^1.0.5"
  },
  "peerDependencies": {
    "react": "^15.4.1 || ^16.3.2"
  },
  "scripts": {
    "build": "hig-scripts-build",
    "build-icons": "node scripts/build-icons/index.js"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  },
  "release": {
    "extends": "@hig/semantic-release-config"
  },
  "babel": {
    "env": {
      "test": {
        "presets": [
          "@hig/babel-preset/test"
        ]
      }
    }
  }
}
