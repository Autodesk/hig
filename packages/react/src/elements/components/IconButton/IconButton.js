/* eslint jsx-a11y/no-noninteractive-tabindex: 0 */

import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import "./icon-button.scss";
import Icon from "../Icon/Icon";

const types = Object.freeze({
  PRIMARY: "primary",
  FLAT: "flat",
  TRANSPARENT: "transparent"
});

const availableTypes = Object.values(types);

export default class IconButton extends Component {
  getTabIndex() {
    return this.props.disabled ? "-1" : "0";
  }

  render() {
    const iconButtonClasses = cx(
      "hig__icon-button",
      `hig__icon-button--${this.props.type}`,
      { "hig__icon-button--disabled": this.props.disabled }
    );

    const props = {
      tabIndex: this.getTabIndex(),
      className: iconButtonClasses,
      title: this.props.title,
      onClick: this.props.onClick,
      onBlur: this.props.onBlur,
      onFocus: this.props.onFocus,
      onMouseEnter: this.props.onMouseEnter
    };

    return this.props.link ? (
      <a {...props} href={this.props.link}>
        <span className="hig__icon-button__icon">
          <Icon
            svg={this.props.svg}
            name={this.props.name}
            nameOrSVG={this.props.icon}
          />
        </span>
      </a>
    ) : (
      <button {...props}>
        <Icon
          svg={this.props.svg}
          name={this.props.name}
          nameOrSVG={this.props.icon}
        />
      </button>
    );
  }
}

IconButton.types = types;

IconButton.defaultProps = {
  type: types.PRIMARY,
  link: null,
  title: "button"
};

IconButton.propTypes = {
  /**
   * Title of the button for accessibility purposes
   */
  title: PropTypes.string.isRequired,
  /**
   * Url button will navigate to when clicked
   */
  link: PropTypes.string,
  /**
   * Name of the icon to be used
   */
  name: PropTypes.oneOf(Icon.AVAILABLE_NAMES),
  /**
   * SVG markup used for the icon
   */
  svg: PropTypes.string,
  /**
   * Deprecated; use `name` or `svg` instead
   */
  icon: PropTypes.string,
  /**
   * Prevents user actions on the button
   */
  disabled: PropTypes.bool,
  /**
   * Called when user moves focus away from the button
   */
  onBlur: PropTypes.func,
  /**
   * Called when user clicks the button
   */
  onClick: PropTypes.func,
  /**
   * Called when user moves focus onto the button
   */
  onFocus: PropTypes.func,
  /**
   * Called when user moves the mouse over the button
   */
  onHover: PropTypes.func,
  /**
   * 'primary' or 'flat'; the style of the button
   */
  type: PropTypes.oneOf(availableTypes)
};
