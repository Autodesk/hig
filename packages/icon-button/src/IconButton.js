/* eslint jsx-a11y/no-noninteractive-tabindex: 0 */

import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import "./icon-button.scss";
import Icon from "@hig/icon";
import { AVAILABLE_NAMES as AVAILABLE_ICON_NAMES } from '@hig/icons';
import types from "./types";
import AVAILABLE_TYPES from './availableTypes';

export default class IconButton extends Component {
  getTabIndex() {
    return this.props.disabled ? "-1" : "0";
  }

  render() {
    const iconButtonClasses = cx(
      "hig__icon-button",
      `hig__icon-button--${this.props.type}`,
      { "hig__icon-button--disabled": this.props.disabled }
    );

    const props = {
      tabIndex: this.getTabIndex(),
      className: iconButtonClasses,
      title: this.props.title,
      onClick: this.props.onClick,
      onBlur: this.props.onBlur,
      onFocus: this.props.onFocus,
      onMouseEnter: this.props.onMouseEnter,
      onMouseLeave: this.props.onMouseLeave
    };

    return this.props.link ? (
      <a {...props} href={this.props.link}>
        <span className="hig__icon-button__icon">
          <Icon
            svg={this.props.svg}
            name={this.props.name}
            nameOrSVG={this.props.icon}
          />
        </span>
      </a>
    ) : (
      <button {...props}>
        <Icon
          svg={this.props.svg}
          name={this.props.name}
          nameOrSVG={this.props.icon}
        />
      </button>
    );
  }
}

IconButton.defaultProps = {
  type: types.PRIMARY,
  link: null,
  title: "button"
};

IconButton.propTypes = {
  /**
   * Title of the button for accessibility purposes
   */
  title: PropTypes.string.isRequired,
  /**
   * Url button will navigate to when clicked
   */
  link: PropTypes.string,
  /**
   * Name of the icon to be used
   */
  name: PropTypes.oneOf(AVAILABLE_ICON_NAMES),
  /**
   * SVG markup used for the icon
   */
  svg: PropTypes.string,
  /**
   * Deprecated; use `name` or `svg` instead
   */
  icon: PropTypes.string,
  /**
   * Prevents user actions on the button
   */
  disabled: PropTypes.bool,
  /**
   * Called when user moves focus away from the button
   */
  onBlur: PropTypes.func,
  /**
   * Called when user clicks the button
   */
  onClick: PropTypes.func,
  /**
   * Called when user moves focus onto the button
   */
  onFocus: PropTypes.func,
  /**
   * Called when user moves the mouse over the button
   */
  onHover: PropTypes.func,
  /**
   * 'primary' or 'flat'; the style of the button
   */
  type: PropTypes.oneOf(AVAILABLE_TYPES)
};
