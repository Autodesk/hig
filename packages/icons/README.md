# Icons

This package retains the source of the library of HIG icons.
You can import individual icons to render on your app.

```jsx
import { Assets24 } from "@hig/icons";

const MyComponent = () => (
    <div className="my-class">
        <Assets24 />
    </div>
);

export default MyComponent
```

Read more about where and how to use icons on [the website](https://hig.autodesk.com/web/basics/icons).

## Getting started

### Install the package

```bash
yarn add @hig/icons @hig/theme-context @hig/theme-data
```

# Developing

## Adding a New Icon

To add a new SVG icon to the library, add the `svg` file to the `src/icons` directory.

Then run the following command:

```bash
yarn build-icons
```

This will use SVGO to optimize the new icon, and update our release files. Remember to commit these changes.
