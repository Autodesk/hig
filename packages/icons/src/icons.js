import "./icons.scss";
import icons from "./release/hig-icons-bundle";
import names from "./release/hig-icon-names";

const AVAILABLE_NAMES = Object.freeze(Object.values(names));

export { icons as default, AVAILABLE_NAMES, names };
