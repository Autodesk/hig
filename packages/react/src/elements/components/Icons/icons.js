import "./icons.scss";

export { default } from "./release/hig-icons-bundle";
export { default as names } from "./release/hig-icon-names";
