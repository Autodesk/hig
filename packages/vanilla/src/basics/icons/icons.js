import './icons.scss';

var IconBundle = require("./release/hig-icons-bundle.js");

module.exports = IconBundle;