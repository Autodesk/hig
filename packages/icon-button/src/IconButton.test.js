import renderer from "react-test-renderer";
import React from "react";

import { names } from "@hig/icons";
import IconButton, { types as iconButtonTypes } from "./IconButton";

describe("IconButton", () => {
  it("has type constants", () => {
    expect(iconButtonTypes).toBeDefined();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<IconButton name={names.SETTINGS} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a link", () => {
    const tree = renderer
      .create(
        <IconButton
          title="Settings"
          link="//example.com"
          name={names.SETTINGS}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with the `icon` prop", () => {
    const tree = renderer.create(<IconButton icon={names.SETTINGS} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
