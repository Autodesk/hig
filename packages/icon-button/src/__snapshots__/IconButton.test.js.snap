// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<button
  className="hig__icon-button hig__icon-button--primary"
  title="Settings"
>
  <svg
    className="hig__icon-button__icon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      settings-24
    </title>
    <path
      d="M21.94 10.85a1 1 0 0 0-.86-1l-1.74-.24-.47-1.13 1.06-1.4a1 1 0 0 0-.09-1.31l-1.63-1.61a1 1 0 0 0-1.31-.08l-1.39 1.06-1.15-.48-.24-1.76a1 1 0 0 0-1-.86h-2.28a1 1 0 0 0-1 .87l-.22 1.76-1.12.47-1.42-1.08a1 1 0 0 0-1.31.09L4.15 5.78a1 1 0 0 0-.08 1.31l1.07 1.4-.48 1.14-1.76.24a1 1 0 0 0-.87 1v2.29a1 1 0 0 0 .86 1l1.77.24.47 1.11-1.08 1.4a1 1 0 0 0 .09 1.32l1.62 1.61a1 1 0 0 0 1.31.09l1.41-1.07 1.13.47.24 1.77a1 1 0 0 0 1 .86h2.29a1 1 0 0 0 1-.87l.23-1.74 1.14-.48 1.4 1.07a1 1 0 0 0 1.32-.09l1.61-1.62a1 1 0 0 0 .09-1.31l-1.06-1.39.48-1.15 1.75-.24a1 1 0 0 0 .86-1zm-3.31 2.62l-.92 2.2 1.41 1.85-1.61 1.62-1.86-1.42-2.21.92-.31 2.36h-2.29l-.32-2.33-2.18-.91-1.87 1.37-1.62-1.61 1.43-1.87-.91-2.18L3 13.15v-2.29l2.33-.31.92-2.2-1.39-1.87 1.61-1.62 1.88 1.42 2.19-.91.3-2.37h2.29l.32 2.32 2.21.92 1.84-1.41 1.62 1.61-1.4 1.91.92 2.19 2.31.31v2.29z"
    />
    <path
      d="M12 8a4 4 0 1 0 4 4 4 4 0 0 0-4-4zm0 7a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
    />
  </svg>
</button>
`;

exports[`IconButton renders correctly with a link 1`] = `
<a
  className="hig__icon-button hig__icon-button--primary"
  href="//example.com"
  tabIndex="0"
  title="Settings"
>
  <svg
    className="hig__icon-button__icon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      settings-24
    </title>
    <path
      d="M21.94 10.85a1 1 0 0 0-.86-1l-1.74-.24-.47-1.13 1.06-1.4a1 1 0 0 0-.09-1.31l-1.63-1.61a1 1 0 0 0-1.31-.08l-1.39 1.06-1.15-.48-.24-1.76a1 1 0 0 0-1-.86h-2.28a1 1 0 0 0-1 .87l-.22 1.76-1.12.47-1.42-1.08a1 1 0 0 0-1.31.09L4.15 5.78a1 1 0 0 0-.08 1.31l1.07 1.4-.48 1.14-1.76.24a1 1 0 0 0-.87 1v2.29a1 1 0 0 0 .86 1l1.77.24.47 1.11-1.08 1.4a1 1 0 0 0 .09 1.32l1.62 1.61a1 1 0 0 0 1.31.09l1.41-1.07 1.13.47.24 1.77a1 1 0 0 0 1 .86h2.29a1 1 0 0 0 1-.87l.23-1.74 1.14-.48 1.4 1.07a1 1 0 0 0 1.32-.09l1.61-1.62a1 1 0 0 0 .09-1.31l-1.06-1.39.48-1.15 1.75-.24a1 1 0 0 0 .86-1zm-3.31 2.62l-.92 2.2 1.41 1.85-1.61 1.62-1.86-1.42-2.21.92-.31 2.36h-2.29l-.32-2.33-2.18-.91-1.87 1.37-1.62-1.61 1.43-1.87-.91-2.18L3 13.15v-2.29l2.33-.31.92-2.2-1.39-1.87 1.61-1.62 1.88 1.42 2.19-.91.3-2.37h2.29l.32 2.32 2.21.92 1.84-1.41 1.62 1.61-1.4 1.91.92 2.19 2.31.31v2.29z"
    />
    <path
      d="M12 8a4 4 0 1 0 4 4 4 4 0 0 0-4-4zm0 7a3 3 0 1 1 3-3 3 3 0 0 1-3 3z"
    />
  </svg>
</a>
`;
